custom_components/foxess_em/__init__.py

```diff
@@ -77,36 +78,48 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
 
     # Check FoxESS API Key before settings up the other platforms to prevent those being setup if it
     # fails, which will prevent setup working when the FoxESS API key is fixed.
-    if len(entry.options) > 0:
-        # overwrite data with options
-        entry.data = entry.options
+    entry_options = copy.deepcopy(dict(entry.options))
+    entry_data = copy.deepcopy(dict(entry.data))
+    _LOGGER.debug("Options %s", entry.options)
+    _LOGGER.debug("Data %s", entry.data)
+
+    if entry_options and entry_options != entry_data:
+        # overwrite data with options, force update as not allowed to access entries directly
+        hass.config_entries.async_update_entry(
+            entry,
+            data=entry_options,
+            version=BatteryManagerFlowHandler.VERSION,
+        )
+        entry_data = copy.deepcopy(dict(entry.options))
+        _LOGGER.debug("Config has been updated")
+    _LOGGER.debug("Data %s", entry.data)
```

Comment:
Isn't this just going to print the exact same thing that was printed on line 84? The value of `entry.data` hasn't changed between these points that I can see. `entry_data` _has_ changed though - was that what you meant to print instead?

Reply:
No `entry.data` and `entry.options` are part of a ConfigEntry - when the config is changed, `entry.options` will be populated and different from `entry.data`, the integration will work correctly as it uses `entry.options` as it's source for it's config data (`entry_data`).

But if you change the config flow again, because `entry.data` hasn't been updated it will pre-populate the config flow with it and so you will see the previous config entry (not the latest); `entry.data` must be updated to correct this. 
As we cannot change config entry items directly since HA core 2024.10.x you have to use async_update_entry which can, and once this has been called `entry.data` == `entry.options` is true (and not the same as line 84).

Reply:
Thanks 👍 

If I knew more about the internals of HA I might be able to review better.

Reply:
No problem it changes so regularly, you feel a bit like a cat chasing a ball of string - you wait for a deprecation warning and then have to interpret the dev notes and implement it before it fails.